packages/itwin/property-grid/src/components/MultiElementPropertyGrid.tsx

```diff
@@ -33,27 +35,32 @@ enum MultiElementPropertyContent {
  * @public
  */
 export interface MultiElementPropertyGridProps extends Omit<PropertyGridProps, "headerControls" | "onBackButton"> {
-  /** Renders controls for ancestors navigation. If set to `undefined` ancestors navigation is disabled. */
+  /** Renders controls for ancestors navigation. If set to `undefined`, ancestors navigation is disabled. */
   ancestorsNavigationControls?: (props: AncestorsNavigationControlsProps) => ReactNode;
 }
 
 /**
  * Component that renders property grid for instances in `UnifiedSelection`.
- * - If multiple instances are selected list containing select instances can be opened that allows to check properties of specific instance.
- * - If single instance is selected navigation through it's ancestors can be enabled.
+ * - If multiple instances are selected, a list containing the selected instances can be opened that allows to check properties of a specific instance.
+ * - If a single instance is selected, navigation through its ancestors can be enabled.
  * @public
  */
 export function MultiElementPropertyGrid({ ancestorsNavigationControls, ...props }: MultiElementPropertyGridProps) {
   const { selectedKeys, focusedInstanceKey, focusInstance, ancestorsNavigationProps } = useInstanceSelection({ imodel: props.imodel });
-
   const [content, setContent] = useState<MultiElementPropertyContent>(MultiElementPropertyContent.PropertyGrid);
+  const { onFeatureUsed } = useTelemetryContext();
 
   useEffect(() => {
     // show standard property grid when selection changes
     return Presentation.selection.selectionChange.addListener(() => {
       setContent(MultiElementPropertyContent.PropertyGrid);
     });
-  }, []);
+  }, [setContent]);
+
+  useEffect(() => {
```

Comment:
Can we instead of `useEffect` wrap `setContent` to report feature usage?

Reply:
On the second though, we probably don't want to wrap setContent, and just call `onFeatureUsed` before `setContent`. In that case we know what feature to report and don't need additional checks that are done in `getFeatureFromContent`

Reply:
The reason it is currently in the `useEffect` is to report the initial content (before any `setContent` is called) and when either the content or selection changes. If we call `onFeatureUsed` before each `setContent` we would still need to handle the other cases.

Reply:
Moved the reporting closer to `setContent`.